### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by M-Anwar-Hussaini**
# 🚀New changes and features in this milestone:
- ✅ Create unit tests using `Rspec` for the following classes:
  - `Book`
  - `Person`
  - `Teacher`
  - `Student`
  - `Classroom`
  - `Rental`
- ✅ The test files are located at `root/src/tests`.
- ✅ No linters error.
- ✅ All tests are passed.

## 📝Note:
- In order to run the tests make sure to install the RSpec gem package installed in your machine by running the following command on the terminal.
```sh
gem install rspec
```

- And also navigate to the `tests` directory.
```sh
cd src/tests/
```
- Command for running all the tests:
```sh
rspec *.rb
```

## Screenshots.
- Test per file
image

- Test all files in one command.
image

### CODE REVIEW

File: src/tests/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    it 'should return the book author' do
+      expect(@book.author).to eql 'F. Scott Fitzgerald'
+    end
+  end
+
+  describe '#rentals' do
+    it 'should have an empty rentals array initially' do
+      expect(@book.rentals).to eql([])
+    end
+  end
```
Comment (tresorsawasawa) [inline src/tests/book_spec.rb:35]: - [x] As required for the project to have tests for all the methods in all the classes, kindly ensure that you also created tests to `#add_rental` method in the `Book` class.

File: src/tests/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    end
+
+    context 'when the person is not of age and lacks parent permission' do
+      it 'does not allow using services' do
+        @person.age = 16
+        @person.parent_permission = false
+        expect(@person.can_use_services?).to eq(false)
+      end
+    end
+  end
```
Comment (tresorsawasawa) [inline src/tests/person_spec.rb:60]: - [x] As required for the project to have tests for all the methods in all the classes, kindly ensure that you also add tests to `#add_rental` method in the `Person` class.

File: src/tests/student_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    it 'The correct name of the Student should be "Mahdi"' do
+      expect(student.correct_name).to eq('Mahdi')
+    end
+  end
+
+  describe '#can_use_services?' do
+    it 'Student can not use from services.' do
+      expect(student.can_use_services?).to eq(false)
+    end
+  end
```
Comment (tresorsawasawa) [inline src/tests/student_spec.rb:42]: - [x] As required for the project to have tests for all the methods in all the classes, kindly ensure that you also add tests to `#play_hooky` method in the `Person` class.

- [x] As required for the project to have tests for all the classes that you created while implementing the Decorator design pattern and all methods in all the classes, kindly ensure that you also add tests in the `Nameable` `TrimmerDecorator`, and `CapitalizeDecorator` and all methods inside.

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @M-Anwar-Hussaini, @PabloBona, and @danifromecuador
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All your tests are passing :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @PabloBona @M-Anwar-Hussaini @danifromecuador, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by PabloBona** [comment]
Thanks for your feedback @anagudelogu  !!

**Pull request merged**